src/obj-ignore.c

```diff
@@ -246,9 +246,10 @@ int apply_autoinscription(struct player *p, struct object *obj)
 	const char *note = obj ? get_autoinscription(obj->kind, aware) : NULL;
 
 	/* Remove unaware inscription if aware */
-	if (aware && quark_str(obj->note) && quark_str(obj->kind->note_unaware) &&
-		streq(quark_str(obj->note), quark_str(obj->kind->note_unaware)))
-		obj->note = 0;
+	if (aware && obj->note && obj->kind->note_unaware &&
+		streq(obj->note, obj->kind->note_unaware)) {
+		string_free(obj->note);
+	}
```

Comment:
Also set obj->note to zero or NULL to avoid a double free when the object is released or the inscription is overwritten.